docs/src/contrasts.md

```diff
@@ -0,0 +1,77 @@
+```@meta
+CurrentModule = StatsModels
+```
+
+# Modeling categorical data
+
+To convert categorical data into a numerical representation suitable for
+modeling, `StatsModels` implements a variety of __contrast coding strategies__.
+Each contrast coding strategy maps a categorical vector with $k$ levels onto
+$k-1$ linearly independent model matrix columns.
+
+## How to specify contrast coding
+
+The default contrast coding strategy is `DummyCoding`.  To override this, use
+the `contrasts` argument when constructing a `ModelFrame`:
+
+```julia
+mf = ModelFrame(y ~ 1 + x, df, contrasts = Dict(:x => EffectsCoding()))
+```
+
+To change the contrast coding for one or more variables in place, use
+
+```@docs
+setcontrasts!
+```
+
+## Interface
+
+```@docs
+AbstractContrasts
+ContrastsMatrix
+```
+
+## Contrast coding strategies
+
+```@docs
+DummyCoding
+EffectsCoding
+HelmertCoding
+ContrastsCoding
+```
+
+### Special internal contrasts
+
+```@docs
+FullDummyCoding
+```
+
+## Further details
+
+### Categorical variables in `Formula`s
+
+Generating model matrices from multiple variables, some of which are
+categorical, requires special care.  The reason for this is that rank-$k-1$
+contrasts are appropriate for a categorical variable with $k$ levels when it is
+_redundant_ with lower-order terms: using rank-$k$ will often result in a
+rank-deficient model matrix which leads to a model that can't be identified.  A
+categorical variable in a term is _redundant_ when the term obtained by dropping
```

Comment:
But "dangerous" sounds so fun!

I actually like something like "the variable aliases the remainder of the term". Then we can replace "redundant variable" with "aliasing variable" and "non-redundant" with "non-aliasing". How does that sound?